atc/metric/emit.go

```diff
@@ -36,9 +37,60 @@ type Monitor struct {
 	eventAttributes  map[string]string
 	emissions        chan eventEmission
 	emitterFactories []EmitterFactory
+
+	Databases       []db.Conn
+	DatabaseQueries Counter
+
+	ContainersCreated Counter
+	VolumesCreated    Counter
+
+	FailedContainers Counter
+	FailedVolumes    Counter
+
+	ContainersDeleted Counter
+	VolumesDeleted    Counter
+	ChecksDeleted     Counter
+
+	JobsScheduled  Counter
+	JobsScheduling Gauge
+
+	BuildsStarted Counter
+	BuildsRunning Gauge
+
+	TasksWaiting Gauge
+
+	ChecksFinishedWithError   Counter
+	ChecksFinishedWithSuccess Counter
+	ChecksQueueSize           Gauge
+	ChecksStarted             Counter
+	ChecksEnqueued            Counter
+
+	ConcurrentRequests         map[string]*Gauge
+	ConcurrentRequestsLimitHit map[string]*Counter
 }
 
-var Metrics = &Monitor{}
+var Metrics = &Monitor{
+	DatabaseQueries:            Counter{},
+	ContainersCreated:          Counter{},
+	VolumesCreated:             Counter{},
+	FailedContainers:           Counter{},
+	FailedVolumes:              Counter{},
+	ContainersDeleted:          Counter{},
+	VolumesDeleted:             Counter{},
+	ChecksDeleted:              Counter{},
+	JobsScheduled:              Counter{},
+	JobsScheduling:             Gauge{},
+	BuildsStarted:              Counter{},
+	BuildsRunning:              Gauge{},
+	TasksWaiting:               Gauge{},
+	ChecksFinishedWithError:    Counter{},
+	ChecksFinishedWithSuccess:  Counter{},
+	ChecksQueueSize:            Gauge{},
+	ChecksStarted:              Counter{},
+	ChecksEnqueued:             Counter{},
```

Comment:
All of these can be omitted, since it's just their zero-values - is there a reason you explicitly instantiate them?

Reply:
copy-paste laziness! good catch